internal/backend/runtime/omni/controllers/omni/cluster_machine_config_status_test.go

```diff
@@ -384,6 +384,93 @@ func (suite *ClusterMachineConfigStatusSuite) TestResetUngraceful() {
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewTalosUpgradeStatusController()))
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewClusterStatusController(false)))
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewClusterConfigVersionController()))
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewMachineConfigGenOptionsController()))
+	suite.Require().NoError(suite.runtime.RegisterQController(omnictrl.NewSchematicConfigurationController(&imageFactoryClientMock{})))
+
+	clusterName := "force-destroy"
+	_, machines := suite.createCluster(clusterName, 3, 0)
+	brokenMachineID := machines[2].Metadata().ID()
+
+	for _, m := range machines {
+		machineSvc, err := suite.newServer(m.Metadata().ID())
+		suite.Require().NoError(err)
+
+		statusSnapshot := omni.NewMachineStatusSnapshot(resources.DefaultNamespace, m.Metadata().ID())
+
+		if m.Metadata().ID() == brokenMachineID {
+			statusSnapshot.TypedSpec().Value.MachineStatus = &machine.MachineStatusEvent{ // set the broken machine stuck in the booting stage
+				Stage: machine.MachineStatusEvent_BOOTING,
+			}
+
+			machineSvc.lock.Lock()
+			machineSvc.etcdLeaveClusterHandler = func(context.Context, *machine.EtcdLeaveClusterRequest) (*machine.EtcdLeaveClusterResponse, error) {
+				return nil, errors.New("cannot leave, wasn't part of it")
+			}
+			machineSvc.onReset = func(context.Context, *machine.ResetRequest) (*machine.ResetResponse, error) {
+				_, updateErr := safe.StateUpdateWithConflicts(suite.ctx, suite.state, statusSnapshot.Metadata(), func(res *omni.MachineStatusSnapshot) error { //nolint:contextcheck
+					res.TypedSpec().Value.MachineStatus.Stage = machine.MachineStatusEvent_MAINTENANCE
+
+					return nil
+				})
+
+				return &machine.ResetResponse{}, updateErr
+			}
+			machineSvc.lock.Unlock()
+		} else {
+			statusSnapshot.TypedSpec().Value.MachineStatus = &machine.MachineStatusEvent{ // set the healthy machines to the running stage
+				Stage: machine.MachineStatusEvent_RUNNING,
+			}
+		}
+
+		suite.Require().NoError(suite.state.Create(suite.ctx, statusSnapshot))
+
+		_, err = safe.StateUpdateWithConflicts(suite.ctx, suite.state, resource.NewMetadata(
+			m.Metadata().Namespace(),
+			omni.MachineStatusType,
+			m.Metadata().ID(),
+			resource.VersionUndefined,
+		),
+			func(res *omni.MachineStatus) error {
+				res.TypedSpec().Value.Connected = true
+				res.TypedSpec().Value.Maintenance = false
+				res.TypedSpec().Value.ManagementAddress = unixSocket + machineSvc.address
+				res.TypedSpec().Value.SecureBootStatus = &specs.SecureBootStatus{
+					Enabled: false,
+				}
+				res.TypedSpec().Value.PlatformMetadata = &specs.MachineStatusSpec_PlatformMetadata{
+					Platform: talosconstants.PlatformMetal,
+				}
+
+				return nil
+			},
+		)
+
+		suite.Require().NoError(err)
+	}
+
+	time.Sleep(time.Second * 1)
```

Comment:
Can we avoid having this timeout here? Maybe some more reliable way to wait?

Reply:
actually it was not needed, so I removed it